File: packages/javascript/lib/string/strToCamelCase.ts
import { capitalizeWord } from './_capitalizeWord.js';
import { words } from './words.js';

export function strToCamelCase(str: string): string {
	if (!str) {
		return '';
	}

	return words(str)
		.map((word: string, index: number) => (index === 0 ? word.toLowerCase() : capitalizeWord(word)))
		.join('');
}

File: packages/python/qsu/string/strToCamelCase.py
from ._capitalizeWord import _capitalizeWord
from .words import words


def strToCamelCase(str: str) -> str:
	if not str:
		return ''

	return ''.join(
		word.lower() if index == 0 else _capitalizeWord(word)
		for index, word in enumerate(words(str))
	)
